viewing_party/party.py

```diff
@@ -0,0 +1,158 @@
+def create_movie(movie_title, genre, rating):
+    create_movie_dict = {'title': movie_title, 'genre': genre, 'rating': rating}
+
+    if movie_title and genre and rating:
+        movie_dict = {'title':movie_title, 'genre':genre, 'rating':rating}
+        return movie_dict
+    else:
+        return None
+
+
+def add_to_watched(user_data, movie_title):
+    watched_list = user_data['watched']
+    watched_list.append(movie_title)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie_title):
+    add_watchlist = user_data['watchlist']
+    add_watchlist.append(movie_title)
+    return user_data
+
+
+def watch_movie(user_data, movie_title):
+    movies_watched = user_data['watched']
+    movies_to_watch = user_data['watchlist']
+
+    for movie in movies_to_watch:
+        if movie_title == movie['title']:
+            movies_watched.append(movie)
+            movies_to_watch.remove(movie)
+             
+    return user_data 
+
+### Wave 2
+
+def get_watched_avg_rating(user_data):
+    avg_rating = 0.0
+    rating_sum = 0
+    if user_data["watched"] == 0:
+        return 0 
+    else: 
+        for movie in range(len(user_data['watched'])):
+            rating_sum += user_data['watched'][]['rating']
+    rating_sum/len(user_data['watched'])
+
+    return avg_rating 
+
+
+def get_most_watched_genre(user_data):
+    most_recurring_genre = {}
+    genre_count = 0
+    max_genre = None
+
+    if len(user_data["watched"])== 0:
+        return None 
+    for movie_dict in user_data["watched"]:
+        if movie_dict["genre"] in most_recurring_genre:
+            most_recurring_genre[movie_dict["genre"]] += 1
+        else:
+            most_recurring_genre[movie_dict["genre"]] = 1
+    # max_genre = max(most_recurring_genre)
+    
+    for genre in most_recurring_genre.items():
+        if most_recurring_genre[1] > genre_count:
+            genre_count = most_recurring_genre[1]
+            max_genre = most_recurring_genre[0]
```

Comment:
There's a mismatch in the variable you used for comparing the current count to the max count so far. Here's a minimal change to get things working:

```suggestion
    for genre in most_recurring_genre.items():
        if genre[1] > genre_count:
            genre_count = genre[1]
            max_genre = genre[0]
```

But we can also use tuple unpacking to get the iterated values in a little nicer fashion (instead of indexing into the tuple):
```python
    for genre, count in most_recurring_genre.items():
        if count > genre_count:
            genre_count = count
            max_genre = genre
```